dev/io.openliberty.mcp.internal_fat/fat/src/io/openliberty/mcp/internal/fat/tool/asyncToolApp/AsyncTools.java

```diff
@@ -70,10 +72,10 @@ public CompletionStage<String> asyncCancellationTool(Cancellation cancellation,
         LOG.info("[asyncCancellationTool] Starting");
         toolStatus.signalStarted(latchName);
         return executor.supplyAsync(() -> {
-            int counter = 0;
-            while (counter++ < 20) {
+            long startTime = System.currentTimeMillis();
+            while ((System.currentTimeMillis() - startTime) < POSITIVE_TIMEOUT_MILIS) {
```

Comment:
`System.nanoTime()` should be used when measuring elapsed time.

`System.currentTimeMillis()` is affected by system time changes (such as a syncing the clock with NTP)